HighwayPatrol.h

```diff
@@ -1,11 +1,19 @@
 #pragma once 
-
 #include "Vehicle.h"
+#include "Highway.h"
 
-struct HighwayPatrol : Vehicle 
+struct HighwayPatrol : public Vehicle 
 {
     HighwayPatrol();
+
+    ~HighwayPatrol() override;
+    HighwayPatrol(const HighwayPatrol&);
+    HighwayPatrol& operator = (const HighwayPatrol&);
+
+
     void scanHighway(Highway* h);
 
     void pullOver( Vehicle* v, bool willArrest, Highway* h );
-};
\ No newline at end of file
+
+    std::string type;
```

Comment:
there is no need to set the scope for the vehicle type to that of a member variable.  it is only used in that `pullOver` function, so make it a local variable in that function.
- [ ] revise